### ISSUE TITLE
Module not found: Error: Can't resolve 'host'

### ISSUE THREAD

**Issue by GDreyV** [angular/angular#30312]
Follow the steps described here: https://angular.io/guide/quickstart
Literally run:
npm install -g @angular/cli
ng new my-app
Then follow steps described here: https://angular.io/guide/using-libraries#library-typings
create src/typings.d.ts with suggested content, add 'host' usage into src/app/app.component.ts.
Return back to the root and run npm start.
As the result I got:
ERROR in ./src/app/app.component.ts
Module not found: Error: Can't resolve 'host' in '.../my-app/src/app'.
Repository with the code: https://github.com/GDreyV/ng-custom-types
What am I doing wrong? Did I miss something? Before it worked with src/typings folder but doesn't work anymore :(

**Comment by underfin** [comment]
@GDreyV . The typeings.d.ts is used for define unknown the `type` with js package or js module. The error caused by `host` package is not exit.
 You can add `/src/app/host.js` 
```
export function parse(url, queryString) {
    return {
        protocol: 'https:',
        hostname: 'angular.io',
        pathname: '/'
    }
}
```
and change `/src/app/app.component.ts`
```
import * as host from './host';
```
I hoped it worked for you.

**Comment by GDreyV** [comment]
Thanks for idea. Really in my original case modules which I'm trying to document is some external module loaded with a simple <script src="..." /> (so it's not a npm package) but which still uses import notation for loading.
I'm not sure, maybe it's webpack packaging error. I thought I could use something like webpack.IgnorePlugin for such cases but I don't see any documentation for overriding webpack settings.

**Comment by underfin** [comment]
It`s typescript compiler error, https://www.typescriptlang.org/docs/handbook/module-resolution.html,you can look it.

**Comment by GDreyV** [comment]
Small summary. As I understand Angular documentation regarding custom typings covers most common case when that library is a npm package.
In my case library was loaded from cdn but still uses commonjs notation. So I have to use "require" to load it. Correct me if I'm wrong but in angular user can't add mixins into webpack configurations for such cases. But there are some plugins which let implement such flow, e.g. ngx-build-plus or Angular-RU/angular-cli-webpack (more details in https://medium.jonasbandi.net/angular-cli-and-moment-js-a-recipe-for-disaster-and-how-to-fix-it-163a79180173 ). 
After that I just added following code into webpack.extra.js:
`const webpack = require('webpack');
module.exports = {
	externals:{ extlib: "commonjs extlib" }
}`
and built it with
`npm run build -- --prod --extra-webpack-config webpack.extra.js --output-hashing=none`
